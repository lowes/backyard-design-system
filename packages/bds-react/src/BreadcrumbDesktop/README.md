# @lowes-tech/bds-react/BreadcrumbDesktop

## Interfaces

- [BreadcrumbDesktopProps](interfaces/BreadcrumbDesktopProps.md)
- [LinkCrumbProps](interfaces/LinkCrumbProps.md)

## References

### default

Renames and re-exports [BreadcrumbDesktop](README.md#breadcrumbdesktop)

## Type aliases

### BreadcrumbDesktopRef

Ƭ **BreadcrumbDesktopRef**: `HTMLElement`

## Variables

### BreadcrumbDesktop

• **BreadcrumbDesktop**: `BDSForwardRef`<[`BreadcrumbDesktopProps`](interfaces/BreadcrumbDesktopProps.md)\>

___

### CrumbDesktop

• **CrumbDesktop**: `BDSFunctional`<`LinkProps`\>
