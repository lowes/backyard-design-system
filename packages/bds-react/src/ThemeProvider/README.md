# @lowes-tech/bds-react/ThemeProvider

## Namespaces

- [ThemeProvider](modules/ThemeProvider.md)

## Interfaces

- [BackyardTheme](interfaces/BackyardTheme.md)
- [ThemeProviderProps](interfaces/ThemeProviderProps.md)

## References

### default

Renames and re-exports [ThemeProvider](README.md#themeprovider)

## Type aliases

### BaseTheme

Ƭ **BaseTheme**: typeof `Theme`

## Variables

### ThemeConsumer

• **ThemeConsumer**: `React.Consumer`<[`BackyardTheme`](interfaces/BackyardTheme.md)\> = `StyledComponentsThemeConsumer`

ThemeConsumer same as 'styled-components'

___

### ThemeContext

• **ThemeContext**: `React.Context`<[`BackyardTheme`](interfaces/BackyardTheme.md)\> = `StyledComponentsThemeContext`

ThemeContext same as 'styled-components'

___

### ThemeProvider

• **ThemeProvider**: `BDSFunctional`<[`ThemeProviderProps`](interfaces/ThemeProviderProps.md)\>

Backyard React Theme Provider

 <ThemeProvider theme="light" font="fellix" someOtherContext="value">
     <Grid.Container>
         ...
             <Button />
         ...
     </Grid.Container>
 </ThemeProvider>

Custom Theme Provider wrapper to provide extra context and overrides for Backyard Themes.

## Functions

### dropShadow

▸ `Const` **dropShadow**(`shadow`, `defaultShadow?`): `string`

Converts box-shadow to drop-shadow, if necessary.

Use for `filter` css instead of `box-shadow`.

**`example`**
```
dropShadow(theme.shadows.shadow_04)
 => 'drop-shadow(0px 2px 1px rgba(0, 0, 0, 0.2)) 
         drop-shadow(0px 1px 1px rgba(0, 0, 0, 0.14)) 
         drop-shadow(0px 1px 3px rgba(0, 0, 0, 0.12))'
```

#### Parameters

| Name | Type | Default value | Description |
| :------ | :------ | :------ | :------ |
| `shadow` | `string` | `undefined` | box-shadow |
| `defaultShadow` | `string` | `'none'` | default shadow if none given |

#### Returns

`string`

___

### getShape

▸ `Const` **getShape**(`shapeProp`, `shapeContext`): ``"rounded"`` \| ``"squared"``

Get the shape or shape context in priority

#### Parameters

| Name | Type | Description |
| :------ | :------ | :------ |
| `shapeProp` | `ExtendedShapes` | shape value from props |
| `shapeContext` | ``"rounded"`` \| ``"squared"`` | shape value from context |

#### Returns

``"rounded"`` \| ``"squared"``

___

### splitShadows

▸ `Const` **splitShadows**(`shadows`, `defaultShadow`): `string`[]

Split the shadows in a given string

**`example`**
```
splitShadows('0px 2px 1px rgba(0, 0, 0, 0.2), 0px 1px 1px rgba(0, 0, 0, 0.14), 0px 1px 3px rgba(0, 0, 0, 0.12)')
 => [
     '0px 2px 1px rgba(0, 0, 0, 0.2)',
     '0px 1px 1px rgba(0, 0, 0, 0.14)',
     '0px 1px 3px rgba(0, 0, 0, 0.12)'
 ]
```

#### Parameters

| Name | Type | Description |
| :------ | :------ | :------ |
| `shadows` | `string` | shadow string from backyard |
| `defaultShadow` | `string` | default shadow if none given |

#### Returns

`string`[]

___

### useBackyardTheme

▸ `Const` **useBackyardTheme**(`props?`): [`BackyardTheme`](interfaces/BackyardTheme.md)

#### Parameters

| Name | Type |
| :------ | :------ |
| `props` | `UseBackyardThemeProps` |

#### Returns

[`BackyardTheme`](interfaces/BackyardTheme.md)

___

### validateBackyardTheme

▸ `Const` **validateBackyardTheme**(`theme`, `name`): `boolean`

#### Parameters

| Name | Type |
| :------ | :------ |
| `theme` | [`BackyardTheme`](interfaces/BackyardTheme.md) |
| `name` | `string` |

#### Returns

`boolean`
