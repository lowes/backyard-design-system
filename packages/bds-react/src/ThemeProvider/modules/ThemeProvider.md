# Namespace: ThemeProvider

## Variables

### bdsName

â€¢ **bdsName**: `string`
