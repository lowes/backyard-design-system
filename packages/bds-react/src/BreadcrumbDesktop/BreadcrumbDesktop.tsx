import * as React from 'react'
import classNames from 'classnames'

import type { BackyardBaseProps } from '../utils/typings/BackyardProps'
import { BDSForwardRef, BDSFunctional } from '../utils/typings/BDSComponentProps'
import Link from '../Link'
import type { LinkProps, LinkRef } from '../Link'
import { useBackyardTheme, validateBackyardTheme } from '../ThemeProvider'

import BreadcrumbDesktopBase from './styles/BreadcrumbDesktopBase'

const CrumbDesktop: BDSFunctional<LinkProps> = React.forwardRef<LinkRef, LinkProps>(function Crumb(
    { className, label, ...props },
    ref,
) {
    return (
        <li>
            <Link className={classNames('crumb', 'crumb-desktop', className)} {...props} ref={ref}>
                {label}
            </Link>
        </li>
    )
})

const BreadcrumbDesktop: BDSForwardRef<BreadcrumbDesktopProps> = React.forwardRef<
    BreadcrumbDesktopRef,
    BreadcrumbDesktopProps
>(function Breadcrumb(
    { ariaLabel = 'Breadcrumb', crumbs, crumb: CrumbComponent = CrumbDesktop, className, ...props },
    ref,
) {
    // Get Backyard Theme
    const theme = useBackyardTheme()

    // Validate theme hook
    validateBackyardTheme(theme, 'BreadcrumbDesktop')

    const items = React.useMemo(
        () =>
            crumbs.map((crumb, index) => (
                <CrumbComponent
                    key={index}
                    {...(index === crumbs.length - 1
                        ? { color: 'primary', 'aria-current': 'page' }
                        : { bold: true, underline: 'hover', color: 'interactive' })}
                    {...(crumb as LinkProps)}
                />
            )),
        [crumbs, CrumbComponent],
    )

    return (
        <BreadcrumbDesktopBase
            aria-label={ariaLabel}
            className={classNames('breadcrumb-desktop', className)}
            ref={ref}
            {...props}
        >
            <ol>{items}</ol>
        </BreadcrumbDesktopBase>
    )
})

type BreadcrumbDesktopRef = HTMLElement

interface LinkCrumbProps extends Omit<LinkProps, 'children'> {
    /**
     * Children are option in Link Crumbs
     */
    children?: React.ReactElement
}

interface BreadcrumbDesktopProps extends BackyardBaseProps<BreadcrumbDesktopRef> {
    /**
     * Label for aria to speak
     */
    ariaLabel?: string
    /**
     * Crumb items to render
     */
    crumbs: LinkCrumbProps[]
    /**
     * Component to render crumb with
     * Defaults to `Link` component
     */
    crumb?: React.ComponentClass
    /**
     * React DOM Class
     */
    className?: string
}

CrumbDesktop.bdsName = 'CrumbDesktop'
BreadcrumbDesktop.bdsName = 'BreadcrumbDesktop'

export { BreadcrumbDesktop, CrumbDesktop }

export type { BreadcrumbDesktopProps, BreadcrumbDesktopRef, LinkCrumbProps }

export default BreadcrumbDesktop
