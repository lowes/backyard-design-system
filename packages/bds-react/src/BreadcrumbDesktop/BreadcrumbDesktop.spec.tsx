import * as React from 'react'

import { render } from '../../test-utils'

import ThemeProvider from '../ThemeProvider'

import BreadcrumbDesktop from './BreadcrumbDesktop'

const crumbs = [
    {
        label: 'Appliances',
        to: '/',
        className: 'crumb-1',
    },
    {
        label: 'Refrigerators',
        to: '/',
        className: 'crumb-2',
    },
    {
        label: 'French Door Refrigerators',
        to: '/',
        className: 'crumb-3 currentPage',
    },
]

describe('BreadcrumbDesktop Snapshots', () => {
    test('BreadcrumbDesktop default snapshot', () => {
        const { asFragment } = render(
            <BreadcrumbDesktop crumbs={crumbs} className="custom-class-name" />,
        )

        expect(asFragment()).toMatchSnapshot()
    })

    describe('themes', () => {
        const themes = ['light', 'dark'] as const

        themes.forEach((theme) => {
            test(`BreadcrumbDesktop ${theme} theme snapshot`, () => {
                const { asFragment } = render(
                    <ThemeProvider theme={theme}>
                        <React.Fragment>
                            <BreadcrumbDesktop crumbs={crumbs} className="custom-class-name" />
                        </React.Fragment>
                    </ThemeProvider>,
                )

                expect(asFragment()).toMatchSnapshot()
            })
        })
    })
})
