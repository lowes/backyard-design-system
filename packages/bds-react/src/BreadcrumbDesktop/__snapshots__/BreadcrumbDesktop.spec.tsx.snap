// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BreadcrumbDesktop Snapshots BreadcrumbDesktop default snapshot 1`] = `
<DocumentFragment>
  .c2 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  cursor: pointer;
  -webkit-transition: all 0.2s ease-in-out;
  transition: all 0.2s ease-in-out;
  -webkit-text-decoration: var(--bds-link-style-underline);
  text-decoration: var(--bds-link-style-underline);
}

.c2 .link-label {
  display: inherit;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  width: 100%;
  font-family: var(--bds-link-style-font-family);
  font-weight: var(--bds-font-weight-regular);
}

.c2 .link-label.bold,
.c2 .link-label b,
.c2 .link-label bold,
.c2 .link-label strong {
  font-family: var(--bds-link-style-font-family);
  font-weight: var(--bds-link-style-font-weight);
}

.c2 .link-label .link-start-icon,
.c2 .link-label .link-end-icon {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c2 .link-label .link-start-icon {
  margin-right: var(--bds-sizes-size-8);
}

.c2 .link-label .link-end-icon {
  margin-left: var(--bds-sizes-size-8);
}

.c2,
.c2:visited,
.c2:hover,
.c2:active {
  color: var(--bds-link-style-color);
  -webkit-text-decoration-color: var(--bds-link-style-color);
  text-decoration-color: var(--bds-link-style-color);
}

.c2 .icon path,
.c2:visited .icon path,
.c2:hover .icon path,
.c2:active .icon path,
.c2 .link-start-icon path,
.c2:visited .link-start-icon path,
.c2:hover .link-start-icon path,
.c2:active .link-start-icon path,
.c2 .link-end-icon path,
.c2:visited .link-end-icon path,
.c2:hover .link-end-icon path,
.c2:active .link-end-icon path {
  fill: var(--bds-link-style-color);
}

.c2.size--small .link-label {
  font-size: var(--bds-sizes-size-12);
  line-height: var(--bds-sizes-size-16);
}

.c2.size--small .link-start-icon,
.c2.size--small .link-end-icon {
  width: var(--bds-sizes-size-12);
  height: var(--bds-sizes-size-12);
}

.c2.size--medium .link-label {
  font-size: var(--bds-sizes-size-14);
  line-height: var(--bds-sizes-size-24);
}

.c2.size--medium .link-start-icon,
.c2.size--medium .link-end-icon {
  width: var(--bds-sizes-size-14);
  height: var(--bds-sizes-size-14);
}

.c2:hover {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.c2:focus {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.c2.c2.c2 {
  outline: none;
  vertical-align: bottom;
  white-space: nowrap;
}

.c2.c2.c2::-moz-focus-inner {
  border: 0;
}

.c0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: flex-end;
  -webkit-box-align: flex-end;
  -ms-flex-align: flex-end;
  align-items: flex-end;
}

.c0 ol {
  margin: 0;
  padding-left: 0;
  list-style: none;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c0 li {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c0 li + li::before {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin: 0 0.5rem;
  -webkit-transform: rotate(15deg);
  -ms-transform: rotate(15deg);
  transform: rotate(15deg);
  border-right: 0.1rem solid var(--bds-color-text-primary);
  height: 0.8rem;
  content: '';
}

.c0 a {
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c0 [aria-current='page'] {
  -webkit-text-decoration: none;
  text-decoration: none;
}

@media only screen and (min-width:82rem) {
  .c2.size--small.c1.size--small .link-label {
    font-size: var(--bds-sizes-size-14);
    line-height: var(--bds-sizes-size-24);
  }

  .c2.size--small.c1.size--small .link-start-icon,
  .c2.size--small.c2.size--small .link-end-icon {
    width: var(--bds-sizes-size-14);
    height: var(--bds-sizes-size-14);
  }
}

@media only screen and (min-width:82rem) {
  .c2.size--medium.c1.size--medium .link-label {
    font-size: var(--bds-sizes-size-16);
    line-height: var(--bds-sizes-size-24);
  }

  .c2.size--medium.c1.size--medium .link-start-icon,
  .c2.size--medium.c2.size--medium .link-end-icon {
    width: var(--bds-sizes-size-16);
    height: var(--bds-sizes-size-16);
  }
}

<nav
    aria-label="Breadcrumb"
    class="c0 breadcrumb-desktop custom-class-name"
  >
    <ol>
      <li>
        <a
          class="c1 c2 backyard link size--medium color--interactive crumb crumb-desktop crumb-1"
          href="/"
          style="--bds-link-style-color: hsl(213, 84%, 45%, 1); --bds-link-style-breakpoint: 82rem; --bds-link-style-font-family: Roboto, Helvetica, Arial, sans-serif; --bds-link-style-font-weight: 500; --bds-link-style-underline: none;"
        >
          <span
            class="label link-label bold"
          >
            Appliances
          </span>
        </a>
      </li>
      <li>
        <a
          class="c1 c2 backyard link size--medium color--interactive crumb crumb-desktop crumb-2"
          href="/"
          style="--bds-link-style-color: hsl(213, 84%, 45%, 1); --bds-link-style-breakpoint: 82rem; --bds-link-style-font-family: Roboto, Helvetica, Arial, sans-serif; --bds-link-style-font-weight: 500; --bds-link-style-underline: none;"
        >
          <span
            class="label link-label bold"
          >
            Refrigerators
          </span>
        </a>
      </li>
      <li>
        <a
          aria-current="page"
          class="c1 c2 backyard link size--medium color--primary crumb crumb-desktop crumb-3 currentPage"
          href="/"
          style="--bds-link-style-color: hsl(0, 0%, 0%, 1); --bds-link-style-breakpoint: 82rem; --bds-link-style-font-family: Roboto, Helvetica, Arial, sans-serif; --bds-link-style-font-weight: 500; --bds-link-style-underline: none;"
        >
          <span
            class="label link-label"
          >
            French Door Refrigerators
          </span>
        </a>
      </li>
    </ol>
  </nav>
</DocumentFragment>
`;

exports[`BreadcrumbDesktop Snapshots themes BreadcrumbDesktop dark theme snapshot 1`] = `
<DocumentFragment>
  .c2 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  cursor: pointer;
  -webkit-transition: all 0.2s ease-in-out;
  transition: all 0.2s ease-in-out;
  -webkit-text-decoration: var(--bds-link-style-underline);
  text-decoration: var(--bds-link-style-underline);
}

.c2 .link-label {
  display: inherit;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  width: 100%;
  font-family: var(--bds-link-style-font-family);
  font-weight: var(--bds-font-weight-regular);
}

.c2 .link-label.bold,
.c2 .link-label b,
.c2 .link-label bold,
.c2 .link-label strong {
  font-family: var(--bds-link-style-font-family);
  font-weight: var(--bds-link-style-font-weight);
}

.c2 .link-label .link-start-icon,
.c2 .link-label .link-end-icon {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c2 .link-label .link-start-icon {
  margin-right: var(--bds-sizes-size-8);
}

.c2 .link-label .link-end-icon {
  margin-left: var(--bds-sizes-size-8);
}

.c2,
.c2:visited,
.c2:hover,
.c2:active {
  color: var(--bds-link-style-color);
  -webkit-text-decoration-color: var(--bds-link-style-color);
  text-decoration-color: var(--bds-link-style-color);
}

.c2 .icon path,
.c2:visited .icon path,
.c2:hover .icon path,
.c2:active .icon path,
.c2 .link-start-icon path,
.c2:visited .link-start-icon path,
.c2:hover .link-start-icon path,
.c2:active .link-start-icon path,
.c2 .link-end-icon path,
.c2:visited .link-end-icon path,
.c2:hover .link-end-icon path,
.c2:active .link-end-icon path {
  fill: var(--bds-link-style-color);
}

.c2.size--small .link-label {
  font-size: var(--bds-sizes-size-12);
  line-height: var(--bds-sizes-size-16);
}

.c2.size--small .link-start-icon,
.c2.size--small .link-end-icon {
  width: var(--bds-sizes-size-12);
  height: var(--bds-sizes-size-12);
}

.c2.size--medium .link-label {
  font-size: var(--bds-sizes-size-14);
  line-height: var(--bds-sizes-size-24);
}

.c2.size--medium .link-start-icon,
.c2.size--medium .link-end-icon {
  width: var(--bds-sizes-size-14);
  height: var(--bds-sizes-size-14);
}

.c2:hover {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.c2:focus {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.c2.c2.c2 {
  outline: none;
  vertical-align: bottom;
  white-space: nowrap;
}

.c2.c2.c2::-moz-focus-inner {
  border: 0;
}

.c0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: flex-end;
  -webkit-box-align: flex-end;
  -ms-flex-align: flex-end;
  align-items: flex-end;
}

.c0 ol {
  margin: 0;
  padding-left: 0;
  list-style: none;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c0 li {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c0 li + li::before {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin: 0 0.5rem;
  -webkit-transform: rotate(15deg);
  -ms-transform: rotate(15deg);
  transform: rotate(15deg);
  border-right: 0.1rem solid var(--bds-color-text-primary);
  height: 0.8rem;
  content: '';
}

.c0 a {
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c0 [aria-current='page'] {
  -webkit-text-decoration: none;
  text-decoration: none;
}

@media only screen and (min-width:82rem) {
  .c2.size--small.c1.size--small .link-label {
    font-size: var(--bds-sizes-size-14);
    line-height: var(--bds-sizes-size-24);
  }

  .c2.size--small.c1.size--small .link-start-icon,
  .c2.size--small.c2.size--small .link-end-icon {
    width: var(--bds-sizes-size-14);
    height: var(--bds-sizes-size-14);
  }
}

@media only screen and (min-width:82rem) {
  .c2.size--medium.c1.size--medium .link-label {
    font-size: var(--bds-sizes-size-16);
    line-height: var(--bds-sizes-size-24);
  }

  .c2.size--medium.c1.size--medium .link-start-icon,
  .c2.size--medium.c2.size--medium .link-end-icon {
    width: var(--bds-sizes-size-16);
    height: var(--bds-sizes-size-16);
  }
}

<nav
    aria-label="Breadcrumb"
    class="c0 breadcrumb-desktop custom-class-name"
  >
    <ol>
      <li>
        <a
          class="c1 c2 backyard link size--medium color--interactive crumb crumb-desktop crumb-1"
          href="/"
          style="--bds-link-style-color: hsl(213, 73%, 68%, 1); --bds-link-style-breakpoint: 82rem; --bds-link-style-font-family: Roboto, Helvetica, Arial, sans-serif; --bds-link-style-font-weight: 500; --bds-link-style-underline: none;"
        >
          <span
            class="label link-label bold"
          >
            Appliances
          </span>
        </a>
      </li>
      <li>
        <a
          class="c1 c2 backyard link size--medium color--interactive crumb crumb-desktop crumb-2"
          href="/"
          style="--bds-link-style-color: hsl(213, 73%, 68%, 1); --bds-link-style-breakpoint: 82rem; --bds-link-style-font-family: Roboto, Helvetica, Arial, sans-serif; --bds-link-style-font-weight: 500; --bds-link-style-underline: none;"
        >
          <span
            class="label link-label bold"
          >
            Refrigerators
          </span>
        </a>
      </li>
      <li>
        <a
          aria-current="page"
          class="c1 c2 backyard link size--medium color--primary crumb crumb-desktop crumb-3 currentPage"
          href="/"
          style="--bds-link-style-color: hsl(0, 0%, 100%, 1); --bds-link-style-breakpoint: 82rem; --bds-link-style-font-family: Roboto, Helvetica, Arial, sans-serif; --bds-link-style-font-weight: 500; --bds-link-style-underline: none;"
        >
          <span
            class="label link-label"
          >
            French Door Refrigerators
          </span>
        </a>
      </li>
    </ol>
  </nav>
</DocumentFragment>
`;

exports[`BreadcrumbDesktop Snapshots themes BreadcrumbDesktop light theme snapshot 1`] = `
<DocumentFragment>
  .c2 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  cursor: pointer;
  -webkit-transition: all 0.2s ease-in-out;
  transition: all 0.2s ease-in-out;
  -webkit-text-decoration: var(--bds-link-style-underline);
  text-decoration: var(--bds-link-style-underline);
}

.c2 .link-label {
  display: inherit;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  width: 100%;
  font-family: var(--bds-link-style-font-family);
  font-weight: var(--bds-font-weight-regular);
}

.c2 .link-label.bold,
.c2 .link-label b,
.c2 .link-label bold,
.c2 .link-label strong {
  font-family: var(--bds-link-style-font-family);
  font-weight: var(--bds-link-style-font-weight);
}

.c2 .link-label .link-start-icon,
.c2 .link-label .link-end-icon {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c2 .link-label .link-start-icon {
  margin-right: var(--bds-sizes-size-8);
}

.c2 .link-label .link-end-icon {
  margin-left: var(--bds-sizes-size-8);
}

.c2,
.c2:visited,
.c2:hover,
.c2:active {
  color: var(--bds-link-style-color);
  -webkit-text-decoration-color: var(--bds-link-style-color);
  text-decoration-color: var(--bds-link-style-color);
}

.c2 .icon path,
.c2:visited .icon path,
.c2:hover .icon path,
.c2:active .icon path,
.c2 .link-start-icon path,
.c2:visited .link-start-icon path,
.c2:hover .link-start-icon path,
.c2:active .link-start-icon path,
.c2 .link-end-icon path,
.c2:visited .link-end-icon path,
.c2:hover .link-end-icon path,
.c2:active .link-end-icon path {
  fill: var(--bds-link-style-color);
}

.c2.size--small .link-label {
  font-size: var(--bds-sizes-size-12);
  line-height: var(--bds-sizes-size-16);
}

.c2.size--small .link-start-icon,
.c2.size--small .link-end-icon {
  width: var(--bds-sizes-size-12);
  height: var(--bds-sizes-size-12);
}

.c2.size--medium .link-label {
  font-size: var(--bds-sizes-size-14);
  line-height: var(--bds-sizes-size-24);
}

.c2.size--medium .link-start-icon,
.c2.size--medium .link-end-icon {
  width: var(--bds-sizes-size-14);
  height: var(--bds-sizes-size-14);
}

.c2:hover {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.c2:focus {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.c2.c2.c2 {
  outline: none;
  vertical-align: bottom;
  white-space: nowrap;
}

.c2.c2.c2::-moz-focus-inner {
  border: 0;
}

.c0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: flex-end;
  -webkit-box-align: flex-end;
  -ms-flex-align: flex-end;
  align-items: flex-end;
}

.c0 ol {
  margin: 0;
  padding-left: 0;
  list-style: none;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c0 li {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c0 li + li::before {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin: 0 0.5rem;
  -webkit-transform: rotate(15deg);
  -ms-transform: rotate(15deg);
  transform: rotate(15deg);
  border-right: 0.1rem solid var(--bds-color-text-primary);
  height: 0.8rem;
  content: '';
}

.c0 a {
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c0 [aria-current='page'] {
  -webkit-text-decoration: none;
  text-decoration: none;
}

@media only screen and (min-width:82rem) {
  .c2.size--small.c1.size--small .link-label {
    font-size: var(--bds-sizes-size-14);
    line-height: var(--bds-sizes-size-24);
  }

  .c2.size--small.c1.size--small .link-start-icon,
  .c2.size--small.c2.size--small .link-end-icon {
    width: var(--bds-sizes-size-14);
    height: var(--bds-sizes-size-14);
  }
}

@media only screen and (min-width:82rem) {
  .c2.size--medium.c1.size--medium .link-label {
    font-size: var(--bds-sizes-size-16);
    line-height: var(--bds-sizes-size-24);
  }

  .c2.size--medium.c1.size--medium .link-start-icon,
  .c2.size--medium.c2.size--medium .link-end-icon {
    width: var(--bds-sizes-size-16);
    height: var(--bds-sizes-size-16);
  }
}

<nav
    aria-label="Breadcrumb"
    class="c0 breadcrumb-desktop custom-class-name"
  >
    <ol>
      <li>
        <a
          class="c1 c2 backyard link size--medium color--interactive crumb crumb-desktop crumb-1"
          href="/"
          style="--bds-link-style-color: hsl(213, 84%, 45%, 1); --bds-link-style-breakpoint: 82rem; --bds-link-style-font-family: Roboto, Helvetica, Arial, sans-serif; --bds-link-style-font-weight: 500; --bds-link-style-underline: none;"
        >
          <span
            class="label link-label bold"
          >
            Appliances
          </span>
        </a>
      </li>
      <li>
        <a
          class="c1 c2 backyard link size--medium color--interactive crumb crumb-desktop crumb-2"
          href="/"
          style="--bds-link-style-color: hsl(213, 84%, 45%, 1); --bds-link-style-breakpoint: 82rem; --bds-link-style-font-family: Roboto, Helvetica, Arial, sans-serif; --bds-link-style-font-weight: 500; --bds-link-style-underline: none;"
        >
          <span
            class="label link-label bold"
          >
            Refrigerators
          </span>
        </a>
      </li>
      <li>
        <a
          aria-current="page"
          class="c1 c2 backyard link size--medium color--primary crumb crumb-desktop crumb-3 currentPage"
          href="/"
          style="--bds-link-style-color: hsl(0, 0%, 0%, 1); --bds-link-style-breakpoint: 82rem; --bds-link-style-font-family: Roboto, Helvetica, Arial, sans-serif; --bds-link-style-font-weight: 500; --bds-link-style-underline: none;"
        >
          <span
            class="label link-label"
          >
            French Door Refrigerators
          </span>
        </a>
      </li>
    </ol>
  </nav>
</DocumentFragment>
`;
